pkg/client/fileCheckpointer.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2022 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+type FileCheckpointer struct {
+	path          string
+	blockNumber   uint64
+	transactionID string
+}
+
+type checkpointState struct {
+	BlockNumber   uint64 `json:"blockNumber"`
+	TransactionID string `json:"transactionId"`
+}
+
+func NewFileCheckpointer(path string) (*FileCheckpointer, error) {
+	fileCheckpointer := new(FileCheckpointer)
+	fileCheckpointer.path = path
```

Comment:
What's there is perfectly valid. This syntax is more typical. Up to you if you want to change it.

```suggestion
	fileCheckpointer := &FileCheckpointer{
		path: path,
	}
```